src/Type/Nette/StringsReplaceCallbackClosureTypeExtension.php

```diff
@@ -0,0 +1,142 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Nette;
+
+use Nette\Utils\Strings;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\Reflection\PassedByReference;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\ClosureType;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\Php\RegexArrayShapeMatcher;
+use PHPStan\Type\StaticMethodParameterClosureTypeExtension;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use function array_key_exists;
+use const PREG_OFFSET_CAPTURE;
+use const PREG_UNMATCHED_AS_NULL;
+
+final class StringsReplaceCallbackClosureTypeExtension implements StaticMethodParameterClosureTypeExtension
+{
+
+	/** @var RegexArrayShapeMatcher */
+	private $regexArrayShapeMatcher;
+
+	public function __construct(RegexArrayShapeMatcher $regexArrayShapeMatcher)
+	{
+		$this->regexArrayShapeMatcher = $regexArrayShapeMatcher;
+	}
+
+	public function isStaticMethodSupported(MethodReflection $methodReflection, ParameterReflection $parameter): bool
+	{
+		return $methodReflection->getDeclaringClass()->getName() === Strings::class
+			&& $methodReflection->getName() === 'replace'
+			&& $parameter->getName() === 'replacement';
+	}
+
+	public function getTypeFromStaticMethodCall(MethodReflection $methodReflection, StaticCall $methodCall, ParameterReflection $parameter, Scope $scope): ?Type
+	{
+		$args = $methodCall->getArgs();
+		$patternArg = $args[1] ?? null;
+
+		if ($patternArg === null) {
+			return null;
+		}
+
+		$replacementType = $scope->getType($args[2]->value);
```

Comment:
This will crash for a call with less than 3 arguments. You need to check if it exists as well.